src/main/java/org/apache/commons/text/TokenFormatterFactory.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.commons.text;
+
+import org.apache.commons.lang3.StringUtils;
+
+public class TokenFormatterFactory {
+
+    /**
+     * Token formatter that returns the token as is.
+     */
+    public static class NoOpFormatter implements TokenFormatter {
+        @Override
+        public String format(char[] prior, int tokenIndex, char[] token) {
+            return new String(token);
+        }
+
+    }
+
+    /**
+     * Token formatter that always returns a constant string, and optionally checks the passed in token
+     * for the constant and throws an error when found.
+     */
+    public static class ConstantTokenFormatter implements TokenFormatter {
+
+        /**
+         * The constant to return.
+         */
+        private char[] constant;
+
+        /**
+         * Whether or not to throw an exception if the constant is found.
+         */
+        private boolean failOnConstantFound = true;
+
+        public ConstantTokenFormatter(char constant) {
+            this(new char[] {constant}, true);
+        }
+
+        public ConstantTokenFormatter(char constant, boolean failOnConstantFound) {
+            this(new char[] {constant}, failOnConstantFound);
+        }
+
+        public ConstantTokenFormatter(String constant) {
+            this(constant, true);
+        }
+
+        public ConstantTokenFormatter(String constant, boolean failOnConstantFound) {
+            this(constant.toCharArray(), failOnConstantFound);
+        }
+
+        public ConstantTokenFormatter(char[] constant, boolean failOnConstantFound) {
+            this.constant = constant;
+            this.failOnConstantFound = failOnConstantFound;
+        }
+
+        @Override
+        public String format(char[] prior, int tokenIndex, char[] token) {
+            if (failOnConstantFound) {
+                for (int i = 0; i < token.length; i++) {
```

Comment:
I think you can exit early once you get within token.length - 1 characters of the end. That is, no need to check the last 16 characters of token for a 17 character constant.